ezflow/data/dataloader/dataloader_creator.py

```diff
@@ -361,14 +374,33 @@ def get_dataloader(self):
             for i in range(len(self.dataset_list) - 1):
                 dataset += self.dataset_list[i + 1]
 
-        data_loader = DataLoader(
-            dataset,
-            batch_size=self.batch_size,
-            pin_memory=self.pin_memory,
-            shuffle=self.shuffle,
-            num_workers=self.num_workers,
-            drop_last=self.drop_last,
-        )
+        if self.distributed:
+
+            sampler = DistributedSampler(
+                dataset,
+                num_replicas=self.world_size,
+                rank=self.rank,
+                shuffle=self.shuffle,
+                drop_last=self.drop_last,
+            )
+            data_loader = DataLoader(
+                dataset,
+                batch_size=self.batch_size,
```

Comment:
I think the batch size should be set based on the number of GPU devices. For example: if `batch_size` is 10 and total GPU devices are 4 then the batch_size for distributed training is 10*4=40.

Reply:
I am not entirely sure about this either. 

This is Professor Huaizu's implementation: [https://github.com/playerkk/deeplab_flow/blob/30a7a758ea75ef4164918b892ee288b71ee44820/tools/train_optical_flow.py#L204](https://github.com/playerkk/deeplab_flow/blob/30a7a758ea75ef4164918b892ee288b71ee44820/tools/train_optical_flow.py#L204)